### DISCUSSION TITLE
Multiprocessing with IfcOpenShell

### DISCUSSION THREAD

**Discussion by DaivaMarcinkeviciute** [IfcOpenShell/IfcOpenShell#5250] [Q&A]
Based on the threads I've found so far (see below), I have the impression that the only way to use IfcOpenShell with Multiprocessing is through the geometry iterator. Is that correct?

Or are there ways to apply multiprocessing to other aspects as well, such as reading and writing IFC files?

Related Threads:
- https://github.com/IfcOpenShell/IfcOpenShell/issues/1660
- https://wiki.osarch.org/index.php?title=IfcOpenShell_code_examples (example with "geometry iterator")
- https://github.com/IfcOpenShell/IfcOpenShell/issues/2056

**Comment by Moult** [comment]
I believe so. However you can always spawn multiple processes.

**Reply by DaivaMarcinkeviciute** [reply]
Thanks for your answer. Maybe i am missing something here, but so far i understood from the afore mentioned threads (for example see [this one](https://sourceforge.net/p/ifcopenshell/discussion/1782716/thread/141c96cc20/)), spawning multiple processes with IfcOpenShell objects produces the error "can't pickle swigpyobject objects". 
I have also tried it myself and ran into the same error. 

Could you please clarify, if you know a way to spawn multiple processes with IfcOpenShell objects and not run into this error?

**Comment by aothms** [comment]
Are you developing in Python or C++?

In anticipation of Py 3.13 which will slowly roll out options to disable the gil, I've been making the file class a bit more thread friendly. I think it's all a bit speculative still how things will evolve, but the future does look pretty interesting.

**Reply by DaivaMarcinkeviciute** [reply]
Thanks for your answer. I am developing in Python. I am excited for the future developments, as it would be great to have more options for multiprocessing!
